[CI] ApproximationGoldenTests testApproximationDisabled failing

**Build Scans:**
- [elasticsearch-intake #37708 / part3](https://gradle-enterprise.elastic.co/s/xcc77vp2r73wc)
- [elasticsearch-pull-request #137830 / part-3](https://gradle-enterprise.elastic.co/s/opqx6vtt4qmhm)
- [elasticsearch-pull-request #137812 / part-3](https://gradle-enterprise.elastic.co/s/45h2y5drj2h24)
- [elasticsearch-pull-request #137812 / part-3](https://gradle-enterprise.elastic.co/s/47megrqn7yjpq)
- [elasticsearch-pull-request #137807 / part-3](https://gradle-enterprise.elastic.co/s/4eh6535qk5iiw)
- [elasticsearch-pull-request #137807 / part-3](https://gradle-enterprise.elastic.co/s/4ae3fpd4rpmly)
- [elasticsearch-pull-request #137765 / part-3](https://gradle-enterprise.elastic.co/s/ousk7dpmrj3q4)
- [elasticsearch-pull-request #137765 / part-3](https://gradle-enterprise.elastic.co/s/etqcwygb6d7bm)
- [elasticsearch-pull-request #137757 / part-3](https://gradle-enterprise.elastic.co/s/wnq4apq5srfw4)
- [elasticsearch-pull-request #137757 / part-3](https://gradle-enterprise.elastic.co/s/oydeurwbllo3m)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:test" --tests "org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests.testApproximationDisabled" -Dtests.seed=A01E720527474D45 -Dtests.locale=ro -Dtests.timezone=America/Recife -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testApproximationDisabled'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Output for test 'testApproximationDisabled' does not match for stages '[LOGICAL_OPTIMIZATION, PHYSICAL_OPTIMIZATION, LOCAL_PHYSICAL_OPTIMIZATION]'
```

**Issue Reasons:**
- [main] 2 consecutive failures in test testApproximationDisabled
- [main] 1 consecutive failures in step part3
- [main] 1 consecutive failures in step part-3
- [main] 1 consecutive failures in pipeline elasticsearch-intake
- [main] 1 consecutive failures in pipeline elasticsearch-pull-request
- [main] 10 failures in test testApproximationDisabled (12.0% fail rate in 83 executions)
- [main] 9 failures in step part-3 (17.6% fail rate in 51 executions)
- [main] 5 failures in pipeline elasticsearch-pull-request (11.1% fail rate in 45 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -289,9 +289,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.PushExpressionToLoadIT
   method: testRoundToLongExactMatch
   issue: https://github.com/elastic/elasticsearch/issues/146615
-- class: org.elasticsearch.xpack.esql.optimizer.ApproximationGoldenTests
-  method: testApproximationDisabled
-  issue: https://github.com/elastic/elasticsearch/issues/146564
 - class: org.elasticsearch.index.reindex.ReindexRelocationOnShutdownIT
   method: testReindexFailsWhenPitRelocationFails
   issue: https://github.com/elastic/elasticsearch/issues/146650

--- a/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/ApproximationGoldenTests.java
+++ b/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/ApproximationGoldenTests.java
@@ -33,7 +33,7 @@ public class ApproximationGoldenTests extends GoldenTestCase {
         runGoldenTest("""
             SET approximation=false;
             FROM many_numbers
-              | STATS SUM(sv), MEDIAN(sv)
+              | STATS COUNT(sv), MEDIAN(sv)
             """, STAGES);
     }
 

--- a/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/local_physical_optimization.expected
+++ b/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/local_physical_optimization.expected
@@ -1,7 +1,7 @@
 LimitExec[1000[INTEGER],null]
-\_AggregateExec[[],[SUM(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],compensated[KEYWORD],long_overflow_throw[KEYWORD]) AS SUM(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],FINAL,[$$SUM(sv)$sum{r}#3, $$SUM(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],null]
-  \_ExchangeExec[[$$SUM(sv)$sum{r}#3, $$SUM(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],true]
-    \_AggregateExec[[],[SUM(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],compensated[KEYWORD],long_overflow_throw[KEYWORD]) AS SUM(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],INITIAL,[$$SUM(sv)$sum{r}#6, $$SUM(sv)$seen{r}#7, $$MEDIAN(sv)$quart{r}#8],16]
+\_AggregateExec[[],[COUNT(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION]) AS COUNT(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],FINAL,[$$COUNT(sv)$count{r}#3, $$COUNT(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],null]
+  \_ExchangeExec[[$$COUNT(sv)$count{r}#3, $$COUNT(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],true]
+    \_AggregateExec[[],[COUNT(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION]) AS COUNT(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],INITIAL,[$$COUNT(sv)$count{r}#6, $$COUNT(sv)$seen{r}#7, $$MEDIAN(sv)$quart{r}#8],16]
       \_FieldExtractExec[sv{f}#0]<[],[],[]>
         \_EsQueryExec[many_numbers], indexMode[standard], [_doc{f}#9], limit[], sort[] estimatedRowSize[8] queryBuilderAndTags [[QueryBuilderAndTags[query={
   "exists" : {

--- a/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/logical_optimization.expected
+++ b/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/logical_optimization.expected
@@ -1,3 +1,3 @@
 Limit[1000[INTEGER],false,false]
-\_Aggregate[[],[SUM(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],compensated[KEYWORD],long_overflow_throw[KEYWORD]) AS SUM(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2]]
+\_Aggregate[[],[COUNT(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION]) AS COUNT(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2]]
   \_EsRelation[many_numbers][mv{f}#3, sv{f}#0, sv_and_one_mv{f}#4]

--- a/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/physical_optimization.expected
+++ b/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/physical_optimization.expected
@@ -1,6 +1,6 @@
 LimitExec[1000[INTEGER],null]
-\_AggregateExec[[],[SUM(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],compensated[KEYWORD],long_overflow_throw[KEYWORD]) AS SUM(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],FINAL,[$$SUM(sv)$sum{r}#3, $$SUM(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],null]
-  \_ExchangeExec[[$$SUM(sv)$sum{r}#3, $$SUM(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],true]
+\_AggregateExec[[],[COUNT(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION]) AS COUNT(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2],FINAL,[$$COUNT(sv)$count{r}#3, $$COUNT(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],null]
+  \_ExchangeExec[[$$COUNT(sv)$count{r}#3, $$COUNT(sv)$seen{r}#4, $$MEDIAN(sv)$quart{r}#5],true]
     \_FragmentExec[filter=null, estimatedRowSize=0, reducer=[], fragment=[<>
-Aggregate[[],[SUM(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],compensated[KEYWORD],long_overflow_throw[KEYWORD]) AS SUM(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2]]
+Aggregate[[],[COUNT(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION]) AS COUNT(sv)#1, PERCENTILE(sv{f}#0,true[BOOLEAN],PT0S[TIME_DURATION],50[INTEGER]) AS MEDIAN(sv)#2]]
 \_EsRelation[many_numbers][mv{f}#6, sv{f}#0, sv_and_one_mv{f}#7]<>]]

--- a/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/query.esql
+++ b/x-pack/plugin/esql/src/test/resources/org/elasticsearch/xpack/esql/optimizer/golden_tests/ApproximationGoldenTests/testApproximationDisabled/query.esql
@@ -1,3 +1,3 @@
 SET approximation=false;
 FROM many_numbers
-  | STATS SUM(sv), MEDIAN(sv)
+  | STATS COUNT(sv), MEDIAN(sv)

